packages/schematic-utils/src/utils/workspace.ts

```diff
@@ -0,0 +1,133 @@
+import { json, virtualFs, workspaces } from "@angular-devkit/core";
+import { noop, Rule, Tree } from "@angular-devkit/schematics";
```

Comment:
Rule and Tree from `packages/schematic-utils/src/engine/interface.ts`
noop from `packages/schematic-utils/src/utils/runtime/base.ts`